internal/impl/packages.go

```diff
@@ -236,19 +236,20 @@ const (
 	ensure installMode = "ensure"
 )
 
-// ensurePackagesAreInstalled ensures:
+// ensureProjectStateIsCurrent ensures the Devbox project state is up to date.
+// Namely:
 //  1. Packages are installed, in nix-profile or runx.
 //     Extraneous packages are removed (references purged, not uninstalled).
-//  2. Files for devbox shellenv are generated
-//  3. Env-vars for shellenv are computed
-//  4. Lockfile is synced
+//  2. Plugins are installed
+//  3. Files for devbox shellenv are generated
+//  4. The Devbox environment is re-computed, if necessary, and cached
+//  5. Lockfile is synced
 //
 // The `mode` is used for:
 // 1. Skipping certain operations that may not apply.
 // 2. User messaging to explain what operations are happening, because this function may take time to execute.
-// TODO: Rename method since it does more than just ensure packages are installed.
-func (d *Devbox) ensurePackagesAreInstalled(ctx context.Context, mode installMode) error {
-	defer trace.StartRegion(ctx, "ensurePackages").End()
+func (d *Devbox) ensureProjectStateIsCurrent(ctx context.Context, mode installMode) error {
```

Comment:
Could be shortened to `ensureStateIsCurrent`

Reply:
I was a tad concerned we'd have some other state (now, or in the future) that is not handled here and so wanted to qualify `State`. But don't feel strongly. Happy to shorten it.